File: app/src/main/java/org/cuieney/videolife/di/component/AppComponent.java
package org.cuieney.videolife.di.component;


import android.content.Context;

import org.cuieney.videolife.common.api.KyApiService;
import org.cuieney.videolife.common.api.OpApiService;
import org.cuieney.videolife.common.api.WyApiService;
import org.cuieney.videolife.di.module.AppModule;
import org.cuieney.videolife.di.module.RetrofitModule;

import javax.inject.Singleton;

import dagger.Component;

/**
 * Created by cuieney on 16/8/7.
 */

@Singleton
@Component(modules = {AppModule.class, RetrofitModule.class})
public interface AppComponent {
    Context getContext();
    KyApiService getKyApiService();
    WyApiService getWyApiService();
    OpApiService getOpApiService();
}

File: app/src/main/java/org/cuieney/videolife/kotlin/di/component/AppComponent.kt
package org.cuieney.videolife.kotlin.di.component

import android.content.Context
import dagger.Component
import org.cuieney.videolife.common.api.*
import org.cuieney.videolife.kotlin.di.module.AppModule
import org.cuieney.videolife.kotlin.di.module.RetrofitModule
import javax.inject.Singleton

/**
 * Created by cuieney on 2017/5/22.
 */
@Singleton
@Component(modules = arrayOf(AppModule::class, RetrofitModule::class))
interface AppComponent{
     fun getContext(): Context
     fun getKyApiService(): KyApiService
     fun getWyApiService(): WyApiService
     fun getOpApiService(): OpApiService
     fun getKuulaApiService(): KuulaApiService
     fun getVeerApiService(): VeerApiService
}